lib/mini_profiler/profiler.rb

```diff
@@ -109,14 +109,15 @@ def serve_results(env)
     end
 
     def generate_html(page_struct, env, result_json = page_struct.to_json)
-        html = MiniProfiler.share_template.dup
-        html.sub!('{path}', "#{env['RACK_MINI_PROFILER_ORIGINAL_SCRIPT_NAME']}#{@config.base_url_path}")
-        html.sub!('{version}', MiniProfiler::ASSET_VERSION)
-        html.sub!('{json}', result_json)
-        html.sub!('{includes}', get_profile_script(env))
-        html.sub!('{name}', page_struct[:name])
-        html.sub!('{duration}', page_struct.duration_ms.round(1).to_s)
-        html
+      path = "#{env['RACK_MINI_PROFILER_ORIGINAL_SCRIPT_NAME']}#{@config.base_url_path}"
+      version = MiniProfiler::ASSET_VERSION
+      json = result_json
+      includes = get_profile_script(env)
+      name = page_struct[:name]
+      duration = page_struct.duration_ms.round(1).to_s
+
+      template = MiniProfiler.share_template
+      ERB.new(template).result(binding)
```

Comment:
one more thing, what if we memoize ERB.new(template), then we get to reuse it and cut down on parse time no?